api/server/router/build/build_routes.go

```diff
@@ -190,8 +144,9 @@ func (br *buildRouter) postBuild(ctx context.Context, w http.ResponseWriter, r *
 	var (
 		context        builder.ModifiableContext
 		dockerfileName string
+		out            io.Writer
 	)
-	context, dockerfileName, err = daemonbuilder.DetectContextFromRemoteURL(r.Body, remoteURL, createProgressReader)
+	context, dockerfileName, err = builder.DetectContextFromRemoteURL(r.Body, remoteURL, createProgressReader)
```

Comment:
We don't want the router depending on the builder either, but maybe this is good for a start.